### PR TITLE
Renamed pygame.sprite.LayeredDirty.set_timing_treshold to pygame.sprite.LayeredDirty.set_timing_threshold

### PR DESCRIPTION
**Pull request by CAPTAIN1947**
Also added a deprecation warning in the docs and asked to use renamed one.

### CODE REVIEW

File: docs/reST/ref/sprite.rst [hunk 1]
```diff
@@ -637,10 +637,24 @@ Sprites are not thread safe. So lock them yourself if using threads.
       Default is 1000./80 where 80 is the fps I want to switch to full screen
       mode.  This method's name is a typo and should be fixed.
 
+      DEPRECATED: Use set_timing_threshold() instead.
```
Comment (ankith26) [inline docs/reST/ref/sprite.rst:640]: For this old function, only this line is enough, other docs can be removed. Additionally a `.. deprecated:: 2.1.1` line must be added here

File: docs/reST/ref/sprite.rst [hunk 2]
```diff
@@ -637,10 +637,24 @@ Sprites are not thread safe. So lock them yourself if using threads.
...
+   .. method:: set_timing_threshold
+
+      | :sl:`sets the threshold in milliseconds`
+      | :sg:`set_timing_threshold(time_ms) -> None`
+
+      Default is 1000./80 where 80 is the fps I want to switch to full screen
+      mode.  This method's name is a typo and should be fixed.
+
+      :raises TypeError: if ``time_ms`` is not int or float
+
```
Comment (ankith26) [inline docs/reST/ref/sprite.rst:655]: Needs a `.. versionadded:: 2.1.1` line here

File: src_py/sprite.py [hunk 3]
```diff
@@ -1332,6 +1333,27 @@ def set_timing_treshold(self, time_ms):
 
         Raises TypeError if time_ms is not int or float.
 
+        """
+        warn('This function will be removed, use set_timing_threshold function instead')
+        if isinstance(time_ms, (int, float)):
+            self._time_threshold = time_ms
+        else:
+            raise TypeError(
```
Comment (ankith26) [inline src_py/sprite.py:1341]: this part can be replaced with `self.set_time_threshold()` to not have code repetition :)

**Review by ankith26 — CHANGES_REQUESTED** [review]
Thanks for the PR! 👍 
Left some reviews :)

File: docs/reST/ref/sprite.rst [hunk 4]
```diff
@@ -634,12 +634,27 @@ Sprites are not thread safe. So lock them yourself if using threads.
       | :sl:`sets the threshold in milliseconds`
       | :sg:`set_timing_treshold(time_ms) -> None`
 
+      DEPRECATED: Use set_timing_threshold() instead.
+
+      .. deprecated:: 2.1.1
+
+      :raises TypeError: if ``time_ms`` is not int or float
```
Comment (ankith26) [inline docs/reST/ref/sprite.rst:641]: this line could be removed too

Comment (CAPTAIN1947) [reply docs/reST/ref/sprite.rst]: Done.

File: src_py/sprite.py [hunk 5]
```diff
@@ -1332,6 +1333,22 @@ def set_timing_treshold(self, time_ms):
...
         Raises TypeError if time_ms is not int or float.
 
+        """
+        warn("This function will be removed, use set_timing_threshold function instead")
+        self.set_timing_threshold(time_ms)
+
+    def set_timing_threshold(self, time_ms):
+        """set the threshold in milliseconds
+
+        set_timing_treshold(time_ms): return None
```
Comment (ankith26) [inline src_py/sprite.py:1343]: ah needs a fix here too :)

File: src_py/sprite.py [hunk 6]
```diff
@@ -1332,6 +1333,22 @@ def set_timing_treshold(self, time_ms):
 
         Raises TypeError if time_ms is not int or float.
 
+        """
+        warn("This function will be removed, use set_timing_threshold function instead")
```
Comment (ankith26) [inline src_py/sprite.py:1337]: Ah, I just realised that `DeprecationWarning` must be passed as a second argument for this function for the warning to be a deprecation warning

Comment (CAPTAIN1947) [reply src_py/sprite.py]: Done. Thanks

**Review by ankith26 — APPROVED** [review]
LGTM! Thanks for the PR! 👍 🎉

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: docs/reST/ref/sprite.rst [hunk 7]
```diff
@@ -634,12 +634,25 @@ Sprites are not thread safe. So lock them yourself if using threads.
...
+
+      .. ## LayeredDirty.set_timing_treshold ##
+
+   .. method:: set_timing_threshold
+
+      | :sl:`sets the threshold in milliseconds`
+      | :sg:`set_timing_threshold(time_ms) -> None`
+
       Default is 1000./80 where 80 is the fps I want to switch to full screen
       mode.  This method's name is a typo and should be fixed.
```
Comment (Starbuck5) [inline docs/reST/ref/sprite.rst:649]: not anymore.

Comment (Starbuck5) [reply docs/reST/ref/sprite.rst:649]: This whole doc has problems.

> Default is 1000./80 where 80 is the fps I want to switch to full screen mode. This method's name is a typo and should be fixed.

What does that even mean?

> It uses the dirty flag technique and is therefore faster than the pygame.sprite.RenderUpdatesGroup sub-class that tracks dirty updates. if you have many static sprites. It also switches automatically between dirty rect update and full screen drawing, so you do no have to worry what would be faster.

So you do **not** have to worry what would be faster

Comment (CAPTAIN1947) [reply docs/reST/ref/sprite.rst:649]: Okay I will fix these.

**Pull request merged**